### DISCUSSION TITLE
Can't select all from a table

### DISCUSSION THREAD

**Discussion by jmaris** [pocketbase/pocketbase#4653] [Q&A] [answered]
Hey,

I'm trying desperately to get my head around pocketbase. I have a collection called groups, each with an id (pocketbase ID), name (string), name_lang (json) and group_id (integer, unique). I need to get the id and groupid of all the records in the groups collection, and then do some things with it.

I'm using javascript and the following code, but even though there are records in the collection, it returns an empty array:

```
routerAdd("GET", "/custom/test", (c) => {
    const result = arrayOf(new DynamicModel({
        "id": "",
        "group_id": 0,
    }));
    const records = $app.dao().db().select("id", "group_id").from("groups").all(result);
    console.log(records);
    console.log(JSON.stringify(records, null));
    return records;
});

```

Could anyone help me to understand what I am doing wrong? Sorry!

**Comment by ganigeorgiev** [comment, marked as answer]
The query result will be inside the `result`, not in `records`.

Additionally, if `groups` is a regular collection, you can also make use of the builtin record helpers - https://pocketbase.io/docs/js-records/.

**Reply by jmaris** [reply]
Got it! Thanks :)
I initially wanted to use the builtin record helpers, but I couldn't find a way to select all records.
I tried this, but it says the filter isn't valid. 
```
const records = $app.dao().findRecordsByFilter(
    "groups",
    "group_id = *");
```
Sorry for the trouble

**Reply by ganigeorgiev** [reply]
At the moment there is no builtin "findAll" method, but you can also use something like:

```js
const records = arrayOf(new Record);
$app.dao().recordQuery("groups").all(records)
console.log(records.length)
```

**Reply by jmaris** [reply]
Thanks, this is perfect :)
